src/main/java/org/tron/common/runtime/vm/VMTimeBenchmarkUtils.java

```diff
@@ -0,0 +1,376 @@
+      if (name != null) {
+        entryBuilder.setName(name);
+      }
+
+      /* { inputs : optional } since fallback function not requires inputs*/
+      if (null != inputs) {
+        for (int j = 0; j < inputs.size(); j++) {
+          JsonElement inputItem = inputs.get(j);
+          if (inputItem.getAsJsonObject().get("name") == null ||
+              inputItem.getAsJsonObject().get("type") == null) {
+            logger.error("Input argument invalid due to no name or no type!");
+            return null;
+          }
+          String inputName = inputItem.getAsJsonObject().get("name").getAsString();
+          String inputType = inputItem.getAsJsonObject().get("type").getAsString();
+          SmartContract.ABI.Entry.Param.Builder paramBuilder = SmartContract.ABI.Entry.Param
+              .newBuilder();
+          paramBuilder.setIndexed(false);
+          paramBuilder.setName(inputName);
+          paramBuilder.setType(inputType);
+          entryBuilder.addInputs(paramBuilder.build());
+        }
+      }
+
+      /* { outputs : optional } */
+      if (outputs != null) {
+        for (int k = 0; k < outputs.size(); k++) {
+          JsonElement outputItem = outputs.get(k);
+          if (outputItem.getAsJsonObject().get("name") == null ||
+              outputItem.getAsJsonObject().get("type") == null) {
+            logger.error("Output argument invalid due to no name or no type!");
+            return null;
+          }
+          String outputName = outputItem.getAsJsonObject().get("name").getAsString();
+          String outputType = outputItem.getAsJsonObject().get("type").getAsString();
+          SmartContract.ABI.Entry.Param.Builder paramBuilder = SmartContract.ABI.Entry.Param
+              .newBuilder();
+          paramBuilder.setIndexed(false);
+          paramBuilder.setName(outputName);
+          paramBuilder.setType(outputType);
+          entryBuilder.addOutputs(paramBuilder.build());
+        }
+      }
+
+      entryBuilder.setType(getEntryType(type));
+      entryBuilder.setPayable(payable);
+      if (stateMutability != null) {
+        entryBuilder.setStateMutability(getStateMutability(stateMutability));
+      }
+
+      abiBuilder.addEntrys(entryBuilder.build());
+    }
+
+    return abiBuilder.build();
+  }
+
+
+  public static byte[] parseABI(String selectorStr, String params) {
+    if (params == null) {
+      params = "";
+    }
+    byte[] selector = new byte[4];
+    System.arraycopy(Hash.sha3(selectorStr.getBytes()), 0, selector, 0, 4);
+    byte[] triggerData = Hex.decode(Hex.toHexString(selector) + params);
+    return triggerData;
+  }
+
+
+  public static class TVMResult {
+
+    private Runtime runtime;
+    private long duration;
+    private byte[] contractAddress;
+
+    public byte[] getContractAddress() {
+      return contractAddress;
+    }
+
+    public TVMResult setContractAddress(byte[] contractAddress) {
+      this.contractAddress = contractAddress;
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Store a copy of "contractAddress". ([squid:S2384](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2384))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1546&issues=AWXr3w7cOlZ5QoPzoUkD&open=AWXr3w7cOlZ5QoPzoUkD)

[ISSUE_KEY]: AWXr3w7cOlZ5QoPzoUkD